### PR TITLE
[From pretrained] Allow returning local path

### PR DESCRIPTION
**Pull request by patrickvonplaten**
It is a common use case to want to know exactly where the pipeline was downloaded. This PR adds a simply kwarg `return_cached_folder` to see this. It could help with issues such as:

### CODE REVIEW

File: src/diffusers/pipeline_utils.py [hunk 1]
```diff
@@ -585,6 +560,33 @@ def load_module(name, value):
                 f"Keyword arguments {unused_kwargs} are not expected by {pipeline_class.__name__} and will be ignored."
             )
 
+        if low_cpu_mem_usage and not is_accelerate_available():
```
Comment (patrickvonplaten) [inline src/diffusers/pipeline_utils.py:563]: this is not used for downloading the model, but just for loading the classes

**Review by anton-l — APPROVED** [review]
LGTM! This should probably be in the flax pipeline as well, to keep them in sync

**Comment by keturn** [comment]
This gets the job done with minimal disruption to the existing API. But methods that don't have a consistent return type are real hard to type-check and use consistently. `from_pretrained` being a method that takes a whole mob of arguments, sometimes bundled up as `**components` or something, makes things a bit riskier as you might be passing that through without paying too much attention to whether one of them is going to change the return type on you.

Alternatives could be:
- stick the `cached_folder` property on the returned instance somewhere, or perhaps in its `.config`.
- add a new method `from_pretrained_and_stuff` that returns a dataclass or namedtuple with `model` and `cached_folder` attributes.
- have some separate function `find_path_of_pretrained(pretrained_model_or_path, revision, subfolder)` that _only_ returns the Path.

**Comment by patrickvonplaten** [comment]
> This gets the job done with minimal disruption to the existing API. But methods that don't have a consistent return type are real hard to type-check and use consistently. `from_pretrained` being a method that takes a whole mob of arguments, sometimes bundled up as `**components` or something, makes things a bit riskier as you might be passing that through without paying too much attention to whether one of them is going to change the return type on you.
> 
> Alternatives could be:
> 
> * stick the `cached_folder` property on the returned instance somewhere, or perhaps in its `.config`.
> * add a new method `from_pretrained_and_stuff` that returns a dataclass or namedtuple with `model` and `cached_folder` attributes.
> * have some separate function `find_path_of_pretrained(pretrained_model_or_path, revision, subfolder)` that _only_ returns the Path.

Think in the future we could have `from_pretrained(...)` better modularized, but for now this should work. Using `return_cached_folder` will stay an edge case and it's turned off by default - think someone wouldn't just turned it on by accident.

**Pull request merged**